battle-engine.js

```diff
@@ -1295,8 +1293,8 @@ BattlePokemon = (function () {
 		if (!type || type === '???') {
 			return true;
 		}
-		if (this.negateImmunity[type]) return true;
-		if (!(this.negateImmunity['Type'] && type in this.battle.data.TypeChart)) {
+		if (!this.battle.runEvent('NegateImmunity', this, type)) return true;
+		if (this.battle.runEvent('NegateImmunity', this, 'Type') || !(type in this.battle.data.TypeChart)) {
```

Comment:
Hmmm. Since the old `negateImmunity['Type']` was designed to only stop natural immunity and not ability-based, the other solution that came to mind was to have the relevant event handlers run the `Immunity` event inside.

But I guess it's better to have 2 events sometimes than 2 events always.